tests/baselines/reference/strictOptionalProperties1.errors.txt

```diff
@@ -15,6 +15,7 @@ strictOptionalProperties1.ts(75,5): error TS2322: Type '[number, undefined, true
   Type at position 1 in source is not compatible with type at position 1 in target.
     Type 'undefined' is not assignable to type 'string'.
 strictOptionalProperties1.ts(99,7): error TS2375: Type '{ foo: undefined; bar: string; }' is not assignable to type 'InputProps' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
+  Properties not assignable with exactOptionalPropertyTypes: foo
   Types of property 'foo' are incompatible.

```

Comment:
To be honest, this just seems confusing. I don't like the idea of adding an equal-level diagnostic when we can avoid it. Do you already know that the incompatibility is from `exactOptionalPropertyTypes`? If so, just don't report the other error.

Reply:
Thanks for looing. I agree that it is superfluous when the child is an EOPT incompatibility.  The issue is that when the child is not an EOPT incompatibility the parent getting the EOPT flavor implies that it is and does not provide any detail on the reason it is marked as EOPT.

Would this be more palatable if the list of properties were added to the parent message instead?  For example:

```
Argument of type '{ a: undefined; z: undefined; }' is not assignable to parameter of type 'A' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties: a, z
   Property 'b' is missing in type '{ a: undefined; }' but required in type 'A'.(2379)
```

I could also try to split this into two separate errors if that would be preferable.

The other options provide less information and seem more difficult to implement:

1. ensure the reported child is an EOPT incompatibility if there is one
   a. losing the non EOPT incompatibility in the error makes it harder when the fix is that the incorrect target type was specified
2. Do not add the EOPT flavor when the reported child is not an EOPT incompatibility
   b. I would prefer this to 1, but in is inconvenient to fix an error only to have another take its place.